_transcripts/005-rust-1.38.0.md

```diff
@@ -0,0 +1,686 @@
+there's no way we can make this safe, because of the combination of these few
+features here. If you were to do this, it would cause instability. And so while
+it's very unlikely that anybody actually did this, you just can't have that kind
+of hole lying around. So yeah, a case where deprecation was not considered and
+there's no, like, light touch. It was no, this has to go immediately. There's
+just no way this can be safe.
+
+__Jon__: Yeah, we don't have too many of these in Rust land, where we decide
+something has to go away.
+
+__Ben__: Even like the "mem uninit" we mentioned before. Like, even though like,
+it's almost— it's considered in some cases, almost impossible to use correctly,
+if you use it on any kind of generic type. That wasn't even like, you know, hard
+removed, even that we could remove it at any point. But now that we have the
+replacement. But `type_id` was just like, nope, it's gotta go.
+
+__Jon__: The only other one I can really think of is, I remember in the early
+days, there was a `thread::scoped` in Rust. You could pass it a closure to run—
+
+__Ben__: Before 1.0? Was this the—
+
+__Jon__: Yeah, I think it was nightly only for— I forget exactly.
+
+__Ben__: There was one case where we had a thing stabilized for one day. And
+then Huon Wilson was like, hey, I was reading the release notes. And actually,
+you have the wrong type signature for this thing. And so we to actually, like,
+the day after. Someone had just forgotten it was an unsafe API, and somebody had
+meant to write a mutable reference when they wrote a shared reference or
+something. It was like, oops!
+
+__Jon__: Also, `thread::scoped` went away entirely, because there wasn't—
+
+__Ben__: I think this was before 1.0, though, which changes the thing entirely.
+
+__Jon__: Yeah, I think you're right.
+
+__Ben__: Before 1.0 was a primordial time. Things changed quickly.
+
+__Jon__: A giant primordial soup.
+
+__Ben__: Yeah. Those are all the meaningful language changes this time around.
+Or, I guess there was actually was a library change. But we have more library
+changes, so—
+
+__Jon__: Yes, we have some smaller ones. So one of them is— and some of you may
+know about this method from before, but if you have a slice, or anything that
+derefs to a slice, then you can join them with some separators. So the idea here
+is— the example that comes most often to mind is, you have a string or you have
+a slice of strings and you want to join them by some separator. So this is
+usually something like a comma. You want a list to be comma separated as a
+single string, and you can do the same thing for a slice. You have a slice of
+`u8`s and you want to join those many slices of `u8`s by some separator, and
+previously you could only join them by a single separator, so a single `u8`.
+Whereas now, with this new change, you can join by something that is a slice
+itself, so you might join by something like `[0, 1, 2]`.
```

Comment:
I'm not certain that `[0, 1, 2]` is what was meant here.

Reply:
I believe the intention here was that you can now join with a list of values which will be used in sequence. So I think I'd just typeset this as a sequence: `0, 1, 2`

Reply:
OK, done.